tests/Functional/ContentConsultationRenderTest.php

```diff
@@ -0,0 +1,432 @@
+    $this->assertEquals('Legal info text', $content_second_group->getText());
+
+    // Set contact and assert content is updated.
+    $node->set('oe_consultation_contacts', $first_contact);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+    $inpage_nav_expected_values = [
+      'title' => 'Page contents',
+      'list' => [
+        ['label' => 'Details', 'href' => '#details'],
+        ['label' => 'Target audience', 'href' => '#target-audience'],
+        ['label' => 'Why we are consulting', 'href' => '#why-we-are-consulting'],
+        ['label' => 'Respond to the consultation', 'href' => '#respond-to-the-consultation'],
+        ['label' => 'Consultation outcome', 'href' => '#consultation-outcome'],
+        ['label' => 'Additional information', 'href' => '#additional-information'],
+        ['label' => 'Legal notice', 'href' => '#legal-notice'],
+        ['label' => 'Contact', 'href' => '#contact'],
+      ],
+    ];
+    $inpage_nav_assert->assertPattern($inpage_nav_expected_values, $navigation->getOuterHtml());
+    $content_items = $content->findAll('xpath', '/div');
+    $this->assertCount(8, $content_items);
+    $this->assertContactEntityDefaultDisplay($content_items[7], 'first_consultation_contact');
+    // Set two contacts and assert label is updated.
+    $node->set('oe_consultation_contacts', [$first_contact, $second_contact]);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+    $inpage_nav_expected_values = [
+      'title' => 'Page contents',
+      'list' => [
+        ['label' => 'Details', 'href' => '#details'],
+        ['label' => 'Target audience', 'href' => '#target-audience'],
+        ['label' => 'Why we are consulting', 'href' => '#why-we-are-consulting'],
+        ['label' => 'Respond to the consultation', 'href' => '#respond-to-the-consultation'],
+        ['label' => 'Consultation outcome', 'href' => '#consultation-outcome'],
+        ['label' => 'Additional information', 'href' => '#additional-information'],
+        ['label' => 'Legal notice', 'href' => '#legal-notice'],
+        ['label' => 'Contacts', 'href' => '#contact'],
+      ],
+    ];
+    $inpage_nav_assert->assertPattern($inpage_nav_expected_values, $navigation->getOuterHtml());
+
+    // Update date values and assert status and respond to the consultation is
+    // updated.
+    $opening_date = (clone $static_time)->modify('+ 3 days');
+    $node->set('oe_consultation_opening_date', ['value' => $opening_date->format(DateTimeItemInterface::DATE_STORAGE_FORMAT)]);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+    $this->assertStatusValue($content, 'Upcoming');
+    $this->assertOpeningDateValue($content, '20 February 2020');
+    $this->assertDeadlineDateValue($content, '28 February 2020, 01:00 (AEDT)');
+    $this->assertSession()->elementTextContains('css', '.ecl-page-header-core .ecl-page-header-core__meta', 'Consultation | Upcoming');
+
+    // Assert status "Closed".
```

Comment:
We should assert here that the button and the text should not be visible in the DOM.